File: OrdenamientoListaDoble/Utilidades.cpp
#include "Utilidades.h"
#include <algorithm>

std::string Utilidades::generarCorreo(const std::string& nombre, const std::string& sNombre, const std::string& apellido) {
    std::string nombreLower = nombre;
    std::string sNombreLower = sNombre;
    std::string apellidoLower = apellido;
    for (int i = 0; i < nombreLower.length(); i++) {
        nombreLower[i] = tolower(nombreLower[i]);
    }
    for (int i = 0; i < sNombreLower.length(); i++) {
        sNombreLower[i] = tolower(sNombreLower[i]);
    }
    for (int i = 0; i < apellidoLower.length(); i++) {
        apellidoLower[i] = tolower(apellidoLower[i]);
    }
    return std::string(1, nombreLower[0]) + std::string(1, sNombreLower[0]) + apellidoLower + "@espe.edu.ec";
}

File: Listas en python/OrdenamientoListaDoble en python/utilidades.py
"""
********************************************************************************************
 *            UNIVERSIDAD DE LAS FUERZAS ARMADAS ESPE                                       *
 * Proposito:                      Lista Doble                                              *
 * Autor:                          Abner Arboleda, Marcelo Acuña, Christian bonifaz         *
 * Fecha de creacion:              26/12/2024                                               *
 * Fecha de modificacion:          26/12/2024                                               *
 * Materia:                        Estructura de datos                                      *
 * NRC :                           1992                                                     *
 ********************************************************************************************
"""
import os

class Utilidades:
    @staticmethod
    def generarCorreo(nombre, sNombre, apellido):
        nombreLower = nombre.lower()
        sNombreLower = sNombre.lower()
        apellidoLower = apellido.lower()

        correoBase = f"{nombreLower[0]}{sNombreLower[0]}{apellidoLower}@espe.edu.ec"
        correoGenerado = correoBase

        if os.path.exists("personas.txt"):
            with open("personas.txt", 'r') as archivo:
                lineas = archivo.readlines()
                contador = 0
                for linea in lineas:
                    datos = linea.strip().split(';')
                    if len(datos) > 4 and datos[4] == correoGenerado:
                        contador += 1
                        correoGenerado = f"{correoBase.split('@')[0]}{contador}@espe.edu.ec"

        return correoGenerado